src/xunit.performance.api/GenericProfiler.cs

```diff
@@ -0,0 +1,12 @@
+﻿using System;
+
+namespace Microsoft.Xunit.Performance.Api
+{
+    internal static class GenericProfiler
+    {
+        public static void Record(string assemblyFileName, string sessionName, string outputDirectory, Action action, Action<string> collectOutputFilesCallback)
+        {
+            action.Invoke();
```

Comment:
Can you just call action.Invoke() directly from the Harness API?  You control the body of the delegate from which this is called, so nothing has to match the "interface".